devito/data/decomposition.py

```diff
@@ -230,9 +230,21 @@ def convert_index(self, *args, rel=True):
                 elif isinstance(glb_idx, slice):
                     if self.loc_empty:
                         return slice(-1, -3)
-                    glb_idx_min = self.glb_min if glb_idx.start is None else glb_idx.start
-                    glb_idx_max = self.glb_max if glb_idx.stop is None else glb_idx.stop-1
-                    retfunc = lambda a, b: slice(a, b + 1, glb_idx.step)
+                    # NOTE: step should not be None at this point
```

Comment:
this method name should probably be ``convert_glb_to_loc``, while the dual one could be ``convert_loc_to_glb``